fusionSuite/frontend
# misc: Reorganize code, fix and improve things

These are lot of small things I fixed or improved while working on https://github.com/fusionSuite/FusionSuite/issues/41. It will be easier to review this PR commit by commit.

**This PR must be merged before https://github.com/fusionSuite/frontend/pull/27**

Changes proposed in this pull request:

- Reorganize files to have less files in `src/app`
- Make sure the tests fail if Angular detects unknown elements
- Remove some code related to types (not used in the application)
- Fix an issue where the page title was always appearing next to the mouse
- Declare layout components as a module so they can be reused
- Add spacing on the right of alerts icon so the alert title looks better
- Add a "not found" page

How to test the feature manually:

1. run the application and verify there are no errors
2. verify the page title doesn't appear when passing the mouse in the window
3. try to login with wrong credentials and verify that the icon is not too close of the "Error" string
4. go to an URL that doesn't exist and verify that a page shows that the page is not found

Pull request checklist:


_If you think one of the item isn’t applicable to the PR, please check it
anyway and precise `N/A` next to it._

Related to:

- https://github.com/fusionSuite/FusionSuite/issues/41
- https://github.com/fusionSuite/frontend/pull/27
